src/utils/condition-validation.ts

```diff
@@ -76,6 +76,10 @@ export const validateAndParseExpirationDate = (dateStr?: string): DateValidation
 
   const parsedDate = parseDate(dateStr);
 
+  if (parsedDate !== dateStr) {
```

Comment:
No, here we are just checking if the provided date is similar to the date returned
Since the behavior of the extra day in a month changes the date provided by the user